app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala

```diff
@@ -270,6 +271,41 @@ object FormTemplateValidator {
     Monoid.combineAll(List(mrc.result, functionsChecker.result))
   }
 
+  def validateErrorMessageConstraints(
+    formTemplate: FormTemplate,
+    allExpressions: List[ExprWithPath]
+  ): ValidationResult = {
+    val allErrorMessages = errorMessages(formTemplate)
+    lazy val allNoPIIFcIds = noPIIFcIds(formTemplate)
+    val invalidResults = for {
+      ss   <- OptionT.liftF(allErrorMessages)
+      fcId <- OptionT.liftF(ss.interpolations).collect { case FormCtx(fcId) => fcId }
```

Comment:
We need to support arbitrarily complex expression, this is matching only `${fooBar}` expressions. 

Look at `LeafExpr`, it should help with inspection of whole expression.

Reply:
I make use of `referenceInfos`. All "errorMessage" expressions contain `errorMessage` in the subpath.